dbt_subprojects/daily_spellbook/models/_metrics/fees/chains/bitcoin/metrics_bitcoin_gas_fees_daily.sql

```diff
@@ -18,29 +20,43 @@ with prices as (
         where
             symbol = 'BTC'
             and blockchain is null
-            {% if is_incremental() %}
+            {% if is_incremental() or true %}
             and {{ incremental_predicate('day') }}
             {% endif %}
 )
 , bitcoin_fees as (
         select
-            date as block_date
-            , sum(total_fees) as daily_fee
+            block_date
+            , from_base58(input[1][6][1]) as address
+            , sum(fee) as daily_fee
         from
-            {{ source('bitcoin', 'blocks') }}
+            {{ source(blockchain, 'transactions') }}
         where
-            date < cast(date_trunc('day', now()) as date) --exclude current day to match prices.usd_daily
-            {% if is_incremental() %}
-            and {{ incremental_predicate('date') }}
+            regexp_like(input[1][6][1], '^[123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz]+$') --filter out invalid addresses
```

Comment:
the only way i can get `from_base58()` to work on bitcoin addresses is if i apply this filter. the same filter is applied in the `cex_bitcoin.addresses` spell, which is used to populate labels.

i can't explain why this is the case with raw bitcoin data, but seems to be a theme

Reply:
https://github.com/duneanalytics/spellbook/blob/4bf03f7f2bfafa07e9770f789296c66ebda47dbd/dbt_subprojects/daily_spellbook/models/_sector/labels/addresses/institution/identifier/cex/labels_cex_bitcoin.sql#L10

Reply:
correction: the CEX spells which feed labels pipeline today use `cast(address as varbinary)`, likely due to `bech32()` function not being available to support that address type. downside is we likely can't convert back to native bitcoin address (not certain on this)